### PR TITLE
Add locationData to the ClientContext

### PR DESCRIPTION
**Pull request by jrmihalick**
### Issues Fixed
PDPOS-6452

### Summary
- Client location data was not available for retail calculations for the first item added to the transaction in Commerce. This is because the `TransMgmtClient` handles setting the `TransLocationModel` on the transaction, but not until after the transaction has already been created and has the first item added to it.
- This adds the `LocationData` onto the `ClientContext` as a json string so that endpoints have access to the `LocationData`.
- Moved the `LocationData` class to `openpos-util` so that consumers do not have to have a dependency on `openpos-flow`, where it was previously located.

### CODE REVIEW

**Review by r-cochran — APPROVED** [review]

File: openpos-util/src/main/java/org/jumpmind/pos/util/model/LocationData.java [hunk 1]
```diff
@@ -50,6 +50,14 @@ public void setLongitude(String longitude) {
         this.longitude = longitude;
     }
 
+    public static LocationData fromString(String jsonLocationData) {
+        try {
+            return DefaultObjectMapper.defaultObjectMapper().readValue(jsonLocationData, LocationData.class);
```
Comment (mmichalek) [inline openpos-util/src/main/java/org/jumpmind/pos/util/model/LocationData.java:55]: Small concern here is new'ing up a an ObjectMapper on each request. I think the ObjectMapper could be a static field as it's promised to be thread safe.
https://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field

Comment (jrmihalick) [reply openpos-util/src/main/java/org/jumpmind/pos/util/model/LocationData.java:55]: Should we change that in the DefaultObjectMapper class and make it a singleton there?

Comment (mmichalek) [reply openpos-util/src/main/java/org/jumpmind/pos/util/model/LocationData.java:55]: 🤯

Comment (mmichalek) [reply openpos-util/src/main/java/org/jumpmind/pos/util/model/LocationData.java:55]: can't argue with that logic

Comment (mmichalek) [reply openpos-util/src/main/java/org/jumpmind/pos/util/model/LocationData.java:55]: Actually, maybe better to leave as is so callers can tweak the default configuration to their needs without disturbing other usages of that default object mapper.

**Review by mmichalek — APPROVED** [review]

File: openpos-flow/src/main/java/org/jumpmind/pos/core/flow/LocationChangedGlobalActionHandler.java [hunk 2]
```diff
@@ -17,6 +17,8 @@
     public void onLocationChanged(Action action) {
         LocationData locationData = Action.convertActionData(action.getData(), LocationData.class);
         clientLocationService.setLocationData(locationData);
+
+        stateManager.getClientContext().put("locationData", locationData != null ? locationData.toString() : null);
```
Comment (chenson42) [inline openpos-flow/src/main/java/org/jumpmind/pos/core/flow/LocationChangedGlobalActionHandler.java:21]: Aren't there more places than just here you need to set the ClientContext?

Comment (chenson42) [reply openpos-flow/src/main/java/org/jumpmind/pos/core/flow/LocationChangedGlobalActionHandler.java:21]: Like in StateManagerContainer.setCurrentStateManager

Comment (jrmihalick) [reply openpos-flow/src/main/java/org/jumpmind/pos/core/flow/LocationChangedGlobalActionHandler.java:21]: I don't think so?  StateManagerContainer reads that map and sticks the entries onto the ClientContext.  See `StateManagerContainer.setCurrentStateManager(IStateManager)`

Comment (chenson42) [reply openpos-flow/src/main/java/org/jumpmind/pos/core/flow/LocationChangedGlobalActionHandler.java:21]: ah

**Review by chenson42 — APPROVED** [review]

**Pull request merged**